rohmu/object_storage/s3.py

```diff
@@ -543,6 +555,7 @@ def multipart_upload_file_object(
         mp_id = cmu_response["UploadId"]
 
         while True:
+            chunk_size = self.get_multipart_chunk_size(current_chunk_size=chunk_size, current_part_number=part_number)
```

Comment:
I think logging the chunk size change would make sense:

```suggestion
            old_chunk_size = chunk_size
            chunk_size = self.get_multipart_chunk_size(current_chunk_size=chunk_size, current_part_number=part_number)
            if chunk_size != old_chunk_size:
                self.log.info("Updated chunk size from %s to %s at part_number %s", old_chunk_size, chunk_size, part_number)
```